packages/@o3r/rules-engine/src/engine/operator/operators/date-based.operators.ts

```diff
@@ -16,6 +17,71 @@ export const inRangeDate: Operator<Date, [DateInput, DateInput], DateInput> = {
   validateRhs: isValidDateRange
 };
 
+/**
+ * Validate if the provided operatorFactValues contain 'o3rCurrentTime'
+ * @param value the value
+ */
+function containsO3rCurrentTime(value: unknown): value is Record<string, Facts> {
+  if (typeof value !== 'object' || value === null) {
+    return false;
+  }
+
+  for (const key in value) {
+    if (typeof key !== 'string') {
+      return false;
+    }
+  }
+
+  return Object.prototype.hasOwnProperty.call(value, 'o3rCurrentTime');
+}
```

Comment:
This function is over complex and the return type is actually incorrect because not checked

```suggestion
function containsO3rCurrentTime(value: any) {
  return !!value?.o3rCurrentTime;
}
```

Reply:
As discuss, the full fonction may need to be removed